File: AHYEON/14.programmers/pro_삼각달팽이.swift
import Foundation

enum Direction {
    case down
    case up
    case right
}


func solution(_ n:Int) -> [Int] {

  var direction: Direction = .down
  var array: [[Int]] = [] // 달팽이 채우기를 진행할 배열
  var sum = 0
  var oneline = n - 1 == 0 ? 1 : n - 1 // 각 방향당 저장할 숫자의 개수
  var count = 1
  var value = 0
  var result: [Int] = []

  for i in 0..<n {
    array.append([Int](repeating: 0, count: i+1))
    sum += i + 1
  }

  // 각 방향별로 배열에 값을 채워준다
  repeat {
    switch direction {
    case .down:
      for i in 0..<oneline {
        array[i+(value*2)][value] = count
        count += 1
      }
      direction = .right

    case .right:
      for i in 0..<oneline {
        array[n-1-value][i+value] = count
        count += 1
      }
      direction = .up

    case .up:
      for i in value..<oneline+value {
        array[n-1-i][n-1-i-value] = count
        count += 1
      }
      direction = .down
      oneline = oneline - 3 == 0 ? 1 : oneline - 3
      value += 1
    }
  } while count <= sum

  for arr in array {
    for i in arr {
      result.append(i)
    }
  }

  return result
}

File: 염성훈/14.Programmers_lv2_3/pro_삼각 달팽이.py
def solution(n):
    #이등변 직각삼각형만들기
    tri = [[0 for i in range(0, j)] for j in range(1, n + 1)]
    # 숫자 채우기
    # 행
    x = -1
    # 열
    y = 0
    k = 1

    for a in range(n):
        # 열 접근
        for b in range(a, n):
            if a % 3 == 0:
                x += 1
            elif a % 3  == 1:
                y += 1
            elif a % 3 == 2:
                x -= 1
                y -= 1
            tri[x][y] = k
            k += 1

    answer = sum(tri, [])
    return answer
